.cirrus.yml

```diff
@@ -160,3 +160,40 @@ docker_builder:
       --tag ghcr.io/cirruslabs/cirrus-cli:$CIRRUS_TAG \
       --tag ghcr.io/cirruslabs/cirrus-cli:latest \
       .
+
+task:
+  name: Check for lacking "buf generate" invocation
+
+  container:
+    image: bufbuild/buf
+
+  generate_script: buf generate
+  check_script: git diff --exit-code
+
+task:
+  only_if: $CIRRUS_TAG != ''
+  name: buf push (tagged)
+
+  container:
+    image: bufbuild/buf
+
+  login_script: echo $BUF_TOKEN | buf registry login --username $BUF_LOGIN --token-stdin
+  push_script: buf push --tag $CIRRUS_TAG
+
+  env:
+    BUF_LOGIN: edigaryev
+    BUF_TOKEN: ENCRYPTED[b52d145079b1e26201b3c1a69f32a28fd9f9176b4b7fca2af1d193296bb7d12f895b2f36871a5d873ca4b51cf3aff15f]
+
+task:
+  only_if: $CIRRUS_BRANCH != ''
+  name: buf push (branch)
```

Comment:
Do we actually need a branch bush on non default branches? Or even at all? Maybe tags are enough